### ISSUE TITLE
crashes on android

### ISSUE THREAD

**Issue by ethanneff** [hirbod/react-native-volume-manager#48]
### Error
```
Fatal Exception: java.lang.RuntimeException
Unable to resume activity {com.adaptedmind.mobile/com.adaptedmind.mobile.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.reactnativevolumemanager.VolumeManagerModule$VolumeBroadcastReceiver.isRegistered()' on a null object reference
```

```
Caused by java.lang.NullPointerException
Attempt to invoke virtual method 'boolean com.reactnativevolumemanager.VolumeManagerModule$VolumeBroadcastReceiver.isRegistered()' on a null object reference
com.reactnativevolumemanager.VolumeManagerModule.registerVolumeReceiver (VolumeManagerModule.java:73)
```

### Versions
react native 0.76.5 and volume manager 2.0.7

### Code
the only code I am using from this library is `if (Constants.app.platform === 'ios') void enableInSilenceMode();`, so this is odd that it is crashing on android

**Comment by hirbod** [comment]
`enableInSilenceMode` is an iOS only method and returns undefined on Android, the call is unrelated. 

https://github.com/hirbod/react-native-volume-manager/blob/b51f62788c9ba9a5f457d56b481fc331a217c94b/src/module.ts#L166-L174

The issue you're reporting is because of a logic issue in `onHostPause` and `onHostResume`, which does not take into account that you never registered the listener in the first place. 

https://github.com/hirbod/react-native-volume-manager/blob/b51f62788c9ba9a5f457d56b481fc331a217c94b/android/src/main/java/com/reactnativevolumemanager/VolumeManagerModule.java#L342-L371

Will fix it.

**Comment by hirbod** [comment]
Fixed in v2.0.8
